lib/matplotlib/tests/test_font_manager.py

```diff
@@ -40,18 +38,12 @@ def test_score_weight():
             fontManager.score_weight(400, 400))
 
 
-def test_json_serialization():
-    # on windows, we can't open a file twice, so save the name and unlink
-    # manually...
-    try:
-        name = None
-        with tempfile.NamedTemporaryFile(delete=False) as temp:
-            name = temp.name
-        json_dump(fontManager, name)
-        copy = json_load(name)
-    finally:
-        if name and os.path.exists(name):
-            os.remove(name)
+def test_json_serialization(tmpdir):
+    # Can't open a NamedTemporaryFile twice on Windows, so use a temporary
+    # directory instead.
+    path = Path(tmpdir, "fontlist.json")
```

Comment:
As a note for the future, when we depend on pytest 3.9, we can use `tmp_path` here.